### PR TITLE
Create vue2 charts

### PR DESCRIPTION
**Pull request by thabarbados**
### Fix or Enhancement?
Create Charts for Vue2, add tests and sandboxes for these charts. 

- [x] All tests passed

### Environment
- OS: BigSur MacOs
- NPM Version: 8.5.1

### CODE REVIEW

File: legacy/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "../.babelrc"
```
Comment (dangreen) [inline legacy/.babelrc:2]: For what Babel is used for?

File: package.json [hunk 2]
```diff
@@ -38,18 +38,23 @@
...
   },
   "scripts": {
     "prepublishOnly": "pnpm test && pnpm build && del ./package && clean-publish",
     "postpublish": "del ./package",
     "emitDeclarations": "tsc --emitDeclarationOnly",
     "build": "rollup -c & pnpm emitDeclarations",
-    "unit": "jest -c jest.config.json",
+    "unit": "pnpm unit:base && pnpm unit:legacy",
+    "unit:base": "jest -c jest.config.json",
+    "unit:legacy": "cd legacy && pnpm install && pnpm unit",
```
Comment (dangreen) [inline package.json:57]:
```suggestion
    "unit:legacy": "cd legacy && pnpm unit",
```

File: package.json [hunk 3]
```diff
@@ -38,18 +38,23 @@
   ],
   "sideEffects": false,
   "main": "./src/index.ts",
+  "legacy": "./legacy/src/index.js",
   "types": "./dist/index.d.ts",
   "publishConfig": {
     "main": "./dist/index.cjs",
     "module": "./dist/index.js",
+    "legacyCjs": "./legacy/index.cjs",
+    "legacyModule": "./legacy/index.js",
```
Comment (dangreen) [inline package.json:47]: it's not supported fields. Lets move this info into legacy/package.json

File: package.json [hunk 4]
```diff
@@ -38,18 +38,23 @@
   ],
   "sideEffects": false,
   "main": "./src/index.ts",
+  "legacy": "./legacy/src/index.js",
```
Comment (dangreen) [inline package.json:41]: ...

File: tsconfig.json [hunk 5]
```diff
@@ -31,7 +31,8 @@
     "skipLibCheck": true
   },
   "include": [
-    "src"
+    "src",
+    "legacy/src/LegacyCharts.js"
```
Comment (dangreen) [inline tsconfig.json:35]:
```suggestion
    "legacy/src"
```

?

**Review by dangreen — CHANGES_REQUESTED** [review]
All legacy components are under "legacy" dir - maybe we will not use "legacy" prefix for components names?

File: package.json (hunk 2 above)
Comment (thabarbados) [reply package.json]: fixed

File: legacy/.babelrc (hunk 1 above)
Comment (thabarbados) [reply legacy/.babelrc:2]: Jest can't work with Vue modules without Babel

File: package.json (hunk 3 above)
Comment (thabarbados) [reply package.json]: moved

File: package.json (hunk 4 above)
Comment (thabarbados) [reply package.json]: removed

File: legacy/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "publishConfig": {
+    "main": "./legacy/index.cjs",
+    "module": "./legacy/index.js"
+  },
```
Comment (dangreen) [inline legacy/package.json:5]:
```suggestion
  "main": "./legacy/src/index.js",
  "publishConfig": {
    "main": "./legacy/index.cjs",
    "module": "./legacy/index.js"
  },
```

**Review by dangreen — CHANGES_REQUESTED** [review]

File: legacy/package.json (hunk 6 above)
Comment (thabarbados) [reply legacy/package.json:5]: @dangreen with this config, we can't import charts from `vue-chartjs/legacy`

Comment (dangreen) [reply legacy/package.json:5]: @thabarbados this package.json shouldn't be in published package.

let's restrict package files with "files" field

https://github.com/reactchartjs/react-chartjs-2/blob/master/package.json#L31-L33

File: legacy/src/LegacyCharts.js [hunk 7]
```diff
@@ -0,0 +1,232 @@
+import {
```
Comment (dangreen) [inline legacy/src/LegacyCharts.js:1]: LegactCharts -> Charts ?

File: legacy/package.json (hunk 6 above)
Comment (thabarbados) [reply legacy/package.json:6]: Thanks, I missed  `Files` field settings. Fixed.

File: legacy/src/LegacyCharts.js (hunk 7 above)
Comment (thabarbados) [reply legacy/src/LegacyCharts.js]: renamed too

**Review by dangreen — APPROVED** [review]

**Pull request merged**